sdk/python/kfp/components/importer_node.py

```diff
@@ -54,6 +54,14 @@ def pipeline_with_importer():
               reimport=False)
           train(dataset=importer1.output)
     """
+    if issubclass(artifact_class, artifact_types.Artifact
+                 ) and not artifact_class.TYPE_NAME.startswith('system.'):
+        # For artifact classes not under the `system` namespace,
+        # use its TYPE_NAME as-is.
+        type_name = artifact_class.TYPE_NAME
```

Comment:
This is the original PR that @chensun created to resolve a bug in use of GCPC artifact types: [#6702](https://github.com/kubeflow/pipelines/pull/6702).

This PR is meant to resolve the issue for use of the importer with GCPC components in KFP v2. The new pipeline tests in this PR confirms that the importer works for the case of a custom artifact type with a `TYPE_NAME` attribute (GCPC or otherwise).

Will defer to @chensun on the above-mentioned proposal.